spi/control-plane/control-plane-spi/src/main/java/org/eclipse/edc/connector/spi/catalog/CatalogService.java

```diff
@@ -25,6 +25,18 @@ public interface CatalogService {
      *
      * @param providerUrl the url of the provider
      * @return the provider's catalog
+     * @deprecated please use {@link #request(String, String, QuerySpec)}
      */
+    @Deprecated(since = "milestone9")
     CompletableFuture<Catalog> getByProviderUrl(String providerUrl, QuerySpec spec);
+
+    /**
+     * Return the catalog of the passed provider url.
+     *
+     * @param providerUrl the url of the provider.
+     * @param protocol the protocol id string.
+     * @param querySpec the {@link QuerySpec} object.
+     * @return the provider's catalog
+     */
+    CompletableFuture<byte[]> request(String providerUrl, String protocol, QuerySpec querySpec);
```

Comment:
The `InputStream` is not usable bc it would be closed when the `Response` is closed.
I tried using a `OutputStream` in the middle but Jersey kinda don't like it, maybe something we can note for a subsequent issue, at the end the catalog, with pagination won't likely be that big.

Reply:
all good, i suspected something along those lines. maybe a word or two of documentation explaining this situation would be useful.